File: Frontend/dgu_laf/lib/model/found_item.dart
class FoundItem {
  final int foundItemId;
  final int userId;
  final String title;
  final String description;
  final String foundDate;
  final int locationId;
  final String createdAt;

  FoundItem({
    required this.foundItemId,
    required this.userId,
    required this.title,
    required this.description,
    required this.foundDate,
    required this.locationId,
    required this.createdAt,
  });

  factory FoundItem.fromJson(Map<String, dynamic> json) {
    return FoundItem(
      foundItemId: json['found_item_id'],
      userId: json['user_id'],
      title: json['title'],
      description: json['description'],
      foundDate: json['found_date'],
      locationId: json['location_id'],
      createdAt: json['created_at'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'found_item_id': foundItemId,
      'user_id': userId,
      'title': title,
      'description': description,
      'found_date': foundDate,
      'location_id': locationId,
      'created_at': createdAt,
    };
  }
}

File: Backend/dgulaf/src/main/java/com/dgulaf/dgulaf/model/FoundItem.java
package com.dgulaf.dgulaf.model;

import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.Date;

@Data
@Entity
@Table(name = "founditems")
public class FoundItem {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer foundItemId;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Column(nullable = false, length = 100)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(nullable = false)
    private Date foundDate;

    @ManyToOne
    @JoinColumn(name = "location_id", nullable = false)
    private ItemLocation location; // ItemLocation 객체를 통해 locationId를 참조합니다.

    @Column(nullable = false, columnDefinition = "TIMESTAMP DEFAULT CURRENT_TIMESTAMP")
    private LocalDateTime createdAt;
}